### PR TITLE
Fix flaky trade test

### PR DESCRIPTION
**Pull request by Quexington**
Should also be created at release (https://github.com/Chia-Network/chia-blockchain/pull/17508).  This one probably should go in after the other is checkpointed into main for diff minimization.

### CODE REVIEW

**Review by AmineKhaldi — APPROVED** [review]

File: tests/wallet/cat_wallet/test_trades.py [hunk 1]
```diff
@@ -1750,7 +1750,7 @@ async def test_trade_conflict(three_wallets_prefarm):
     await time_out_assert(15, get_trade_and_status, TradeStatus.PENDING_CONFIRM, trade_manager_taker, tr1)
     # pushing into mempool while already in it should fail
     tr2, txs2 = await trade_manager_trader.respond_to_offer(offer, peer, DEFAULT_TX_CONFIG, fee=uint64(10))
-    await trade_manager_taker.wallet_state_manager.add_pending_transactions(txs2)
+    await trade_manager_trader.wallet_state_manager.add_pending_transactions(txs2)
```
Comment (arvidn) [inline tests/wallet/cat_wallet/test_trades.py:1753]: typikcally the counterpart to "taker" is "maker". I imagine that's what "trader" refers to here. is that right?
It's not intuitive why this change makes a difference or why it makes sense. Presumably it's supposed to be the *taker* that submits the transaction

Comment (arvidn) [reply tests/wallet/cat_wallet/test_trades.py:1753]: oh, there are 3 wallets involved here, and there is both a taker and maker.

Comment (Quexington) [reply tests/wallet/cat_wallet/test_trades.py:1753]: Yeah poor naming forsure but the purpose of this change is so that the transactions being created are actually pushed to the correct wallet state manager.

**Review by altendky — DISMISSED** [review]
Just blocking while chatting with Amine to clarify ordering of PRs.

**Review by emlowe — APPROVED** [review]

**Comment by Starttoaster** [comment]
Do we have any more pre-merge changes in flight, or is this one ready for merge? Asking because of the text in the PR body.
